### PR TITLE
Am 112 x debug paths splitting

### PR DESCRIPTION
**Pull request by NTrevisani**
#### PR description:

- splitting the muon paths in two different paths, one considering only hits in SL3 and one considering only hits in SL1;
- considering more than one hit per cell, duplicating muon paths accordingly;
- using patterns dedicated to the specific DT chamber geometry;
- efficiency plots wrt standard grouping divided in wheels.

#### PR validation:

It is possible to run the code with the old configuration: no paths splitting into SL1/SL3, no more than 1 hit per cell considererd, and just one pattern for all DT chambers.
No changes are observed.

### CODE REVIEW

**Comment by NTrevisani** [comment]
To include the changes in the AM_112X_debug branch. It is good to have a review so that I can also prepare a pull request for the AM_112X_dev branch (in that case, I shall take care of include changes that do not affect the data format).

**Comment by NTrevisani** [comment]
Ready for comparison.

**Review by folguera — COMMENTED** [review]
los .root files de los patterns debería de colocarse en otro repo:  https://github.com/dtp2-tpg-am/L1Trigger-DTTriggerPhase2, quítalos porfa.

File: L1Trigger/DTTriggerPhase2/interface/MuonPathAnalyzerInChamber.h [hunk 1]
```diff
@@ -38,7 +38,8 @@ class MuonPathAnalyzerInChamber : public MuonPathAnalyzer {
   void setMinHits4Fit(int h) { minHits4Fit_ = h; };
   void setChiSquareThreshold(float ch2Thr) { chiSquareThreshold_ = ch2Thr; };
   void setMinimumQuality(cmsdt::MP_QUALITY q) {
-    if (minQuality_ >= cmsdt::LOWQGHOST)
+    // if (minQuality_ >= cmsdt::LOWQGHOST)
+    if (minQuality_ >= cmsdt::H3PLUS0)
```
Comment (folguera) [inline L1Trigger/DTTriggerPhase2/interface/MuonPathAnalyzerInChamber.h:42]: este parámetro quizá tenga sentido que sea configurable.

File: L1Trigger/DTTriggerPhase2/src/MuonPathAnalyzerInChamber.cc [hunk 2]
```diff
@@ -541,30 +587,71 @@ void MuonPathAnalyzerInChamber::evaluateQuality(MuonPathPtr &mPath) {
   mPath->setNPrimitivesUp(nPrimsUp);
   mPath->setNPrimitivesDown(nPrimsDown);
   
+
+  // // OLD QUALITY ASSIGNMENT
```
Comment (folguera) [inline L1Trigger/DTTriggerPhase2/src/MuonPathAnalyzerInChamber.cc]: quita estas líneas comentadas porfa

File: L1Trigger/DTTriggerPhase2/interface/MuonPathAnalyzerInChamber.h (hunk 1 above)
Comment (NTrevisani) [reply L1Trigger/DTTriggerPhase2/interface/MuonPathAnalyzerInChamber.h:42]: Hasta ahora estuvo hard-coded y usando la calidad más baja. Como no es trivial hacerlo configurable (me decían Carlos y Sergio que lo más facil sería pasar en input un string y dependiendo del valor del string darle el valor de calidad correspondiente. e.g., input_string = "H3PLUS0" --> minQuality = H3PLUS0) de momento lo dejaría así, Pero como digas.

**Pull request merged**